framework/src/meshgenerators/BreakMeshByBlockGenerator.C

```diff
@@ -28,14 +28,39 @@ BreakMeshByBlockGenerator::validParams()
                              "attached. Naming convention for the new boundaries will be the old "
                              "boundary name plus \"_to_\" plus the subdomain name. At the moment"
                              "this only works on REPLICATED mesh");
+  params.addParam<std::vector<SubdomainID>>(
+      "block",
+      "The list of subdomain names where neml mehcanisc should be used, "
+      "default all blocks.");
+  params.addParam<bool>("add_transition_interface",
+                        true,
+                        "If true (default) and block is not empty, a special boundary named "
+                        "interface_transition is generate between listed blocks and other blocks. "
```

Comment:
This isn't a requirement - but a part of me thinks that we should name this boundary something a little more "unique" - something like `name() + "interface_transition"` in the off-case that someone already has a block with this name. That, or error out of the block `interface_transition` already exists before executing this generator

Reply:
> This isn't a requirement - but a part of me thinks that we should name this boundary something a little more "unique" - something like `name() + "interface_transition"` in the off-case that someone already has a block with this name. That, or error out of the block `interface_transition` already exists before executing this generator

Agree with you that we should check that a boundary with that name does not exist.
How do I check that a boundary with that name already exists in the mesh? In `meshBase` class I didn't find anything useful

Reply:
Should be:
```
get_boundary_info().get_id_by_name("interface_transition") != invalid_id
```
if it exists in the `BoundaryInfo` currently. Not the most intuitive, unfortunately!

Reply:
Thanks also, I added a parameter to provide a user-defined  `interface_transition` name